CRM/Activityreport/Data.php

```diff
@@ -4,33 +4,145 @@
  * Provide static methods to retrieve and format an Activity data.
  */
 class CRM_Activityreport_Data {
-  const ROWS_TO_RETURN = 1000;
+  const ROWS_CACHE_LIMIT = 1000;
+  const ROWS_RETURN_LIMIT = 10000;
+
   private static $fields = array();
   private static $emptyRow = array();
   private static $multiValues = array();
   private static $formattedValues = array();
   private static $customizedValues = array();
 
   /**
-   * Return an array containing formatted Activity data.
+   * Return an array containing formatted Activity data and information
+   * needed to make a call for more data.
    *
-   * @param int $offset
-   *   Offset for API call
-   * @param int $limit
-   *   Limit for API call
-   * @param int $multiValuesOffset
-   *   Multivalues offset
    * @param string $startDate
    *   "Date from" value to filter Activities by their date
    * @param string $endDate
    *   "Date to" value to filter Activities by their date
+   * @param int $page
+   *   Page offset
    *
    * @return array
    */
-  public static function get($offset = 0, $limit = 0, $multiValuesOffset = 0, $startDate = null, $endDate = null) {
+  public static function get($startDate = null, $endDate = null, $page = 0) {
+    $data = array();
+    $nextDate = NULL;
+    $nextPage = NULL;
+    $break = FALSE;
+
+    $query = 'SELECT path, data FROM civicrm_cache WHERE group_name = "pivotreport"';
+    $params = array();
+
+    if (!empty($startDate)) {
+      $query .= ' AND path >= %1';
+      $params[1] = array('data_' . substr($startDate, 0, 10) . '_' . str_pad($page, 6, '0', STR_PAD_LEFT), 'String');
+    }
+
+    if (!empty($endDate)) {
+      $query .= ' AND path <= %2';
+      $params[2] = array('data_' . substr($endDate, 0, 10) . '_999999', 'String');
+    }
+
+    $query .= ' ORDER BY path ASC';
+
+    $cache = CRM_Core_DAO::executeQuery($query, $params);
```

Comment:
Instead of using a raw SQL query, it's better to use the BAO instead/DAO instead. It has some methods like `addWhere` that I think you can use to build the query